state-chain/custom-rpc/src/lib.rs

```diff
@@ -369,15 +384,22 @@ pub trait CustomApi {
 	#[method(name = "pool_price")]
 	fn cf_pool_price(
 		&self,
-		from_asset: RpcAsset,
-		to_asset: RpcAsset,
+		from: RpcAsset,
+		to: RpcAsset,
 		at: Option<state_chain_runtime::Hash>,
-	) -> RpcResult<Option<PoolPrice>>;
+	) -> RpcResult<Option<PoolPriceV1>>;
+	#[method(name = "pool_price_v2")]
+	fn cf_pool_price_v2(
+		&self,
+		base_asset: RpcAsset,
+		quote_asset: RpcAsset,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<PoolPriceV2>;
 	#[method(name = "swap_rate")]
 	fn cf_pool_swap_rate(
 		&self,
-		from_asset: RpcAsset,
-		to_asset: RpcAsset,
+		base_asset: RpcAsset,
```

Comment:
These need to not be change to preserve compatibility. And the "from" and "to" terminology implies a direction, that base and quote does not.